pkg/agent/proxy/proxier.go

```diff
@@ -978,9 +1002,25 @@ func NewProxier(
 		p.endpointsConfig = config.NewEndpointsConfig(informerFactory.Core().V1().Endpoints(), resyncPeriod)
 		p.endpointsConfig.RegisterEventHandler(p)
 	}
+	if p.multiclusterEnabled {
+		// We register event handler here since serviceCIDRIPv4 is only required
+		// when Antrea Multi-cluster is enabled.
+		fnRegisterServiceCIDREventHandler(p)
+	}
 	return p
 }
 
+func registerServiceCIDREventHandler(p *proxier) {
```

Comment:
I think this function could be a method of `proxier`.

Reply:
Even we reset in the test code, it can still be a func of proxier, unless we must pass proxier to the func (do we really need that?) in the test code?

Reply:
Done, re-based the code and refine the test part.